train.py

```diff
@@ -307,8 +308,11 @@ def loss_fn(pred, labels):
 
             # clip gradients
             for m in model_parts:
```

Comment:
I think we shouldn't do this for each m in model_parts. The `total_norm` has to be an overall sum.

Reply:
You are right. This for loop would introduce more problems. If my understanding is correct, `model_parts` contains more than 1 nn.Module only when we are using multi-stage PP. So we can simply just use [`itertools.chain`](https://docs.python.org/3/library/itertools.html#itertools.chain.from_iterable) to chain the `.parameters()` then pass to the clip norm function

Reply:
Pushed a new commit with the above mentioned logic

Reply:
Updated with the newest PyTorch clip norm APIs